out-of-tree build error

out-of-tree build in `build` directory fails since `build/include/clspv` does not exist when writing a python processed file.

```
$ cd <clspv-dir>
$ mkdir build
$ cd build
$ cmake ..
$ make
...
Traceback (most recent call last):
  File "/home/syoyo/work/clspv/cmake/spirv_glsl.py", line 86, in <module>
    main()
  File "/home/syoyo/work/clspv/cmake/spirv_glsl.py", line 53, in main
    with open(args.output_file, "w") as output, open(args.input_file, "r") as input:
FileNotFoundError: [Errno 2] No such file or directory: '/home/syoyo/work/clspv/build/include/clspv/spirv_glsl.hpp'
Traceback (most recent call last):
  File "/home/syoyo/work/clspv/cmake/bake_file.py", line 100, in <module>
    main()
  File "/home/syoyo/work/clspv/cmake/bake_file.py", line 41, in main
    with open(args.output_file, "w") as output:
FileNotFoundError: [Errno 2] No such file or directory: '/home/syoyo/work/clspv/build/include/clspv/opencl_builtins_header.h'
```

The reason is `CLSPV_BINARY_DIR` is set to binary dir(i.e. `build`).
Changing it to `CMAKE_CURRENT_SOURCE_DIR` will solve the issue.
(or create `build/include/clspv` folder in `spirv_glsl.py`)

```
set(CLSPV_BINARY_DIR ${CMAKE_CURRENT_SOURCE_DIR})
```

FYI, in-source build also fails at bootstrapping LLVM.

```
$ cd <clspv-dir>
$ cmake .
CMake Error at third_party/llvm/CMakeLists.txt:231 (message):
  In-source builds are not allowed.

  Please create a directory and run cmake from there, passing the path

  to this source directory as the last argument.

  This process created the file `CMakeCache.txt' and the directory
  `CMakeFiles'.

  Please delete them.
```

--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -17,6 +17,8 @@ set(SPIRV_C_STRINGS_OUTPUT_FILE ${CLSPV_BINARY_DIR}/include/clspv/spirv_c_string
 set(SPIRV_C_STRINGS_NAMESPACE spv)
 set(SPIRV_C_STRINGS_PYTHON_FILE ${CMAKE_CURRENT_SOURCE_DIR}/spirv_c_strings.py)
 
+file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/include/clspv")
+
 add_custom_target(clspv_c_strings
   DEPENDS ${SPIRV_C_STRINGS_OUTPUT_FILE})
 

